### PR TITLE
ci(PG-1288): update license checker tool

### PR DESCRIPTION
**Pull request by pshoniuk**

### CODE REVIEW

**Comment by bejayoharen** [comment]
LGTM. I'm going to let @giannif approve it, though, since I think he's more likely to follow the details.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# editorconfig.org
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[{*.json,*.yaml,*.yml}]
```
Comment (giannif) [inline .editorconfig:12]: @pshoniuk did you want to commit this? not sure if these format rules conflict with prettier?

File: scripts/license-checker.docker.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  'util')
+    sd_app_name='giphy-js-utils'
+    ;;
+  *)
+    sd_app_name="giphy-js-${package_name}"
+    ;;
+  esac
+
+  printf "\033[0;32mChecking dependencies in %s\033[0m\n" "$sd_app_name"
+  docker run --rm \
```
Comment (giannif) [inline scripts/license-checker.docker.sh:19]: @pshoniuk this runs a docker container for every package?

File: scripts/license-checker.js [hunk 3]
```diff
@@ -19,21 +19,24 @@ async function fetchJSON(url) {
...
             .on('error', reject)
     })
 }
 
-async function fetchLicenseCategories(categories) {
-    const data = {}
-    const uniqueCategories = [...new Set([...categories])]
-    for (const category of uniqueCategories) {
-        data[category] = await fetchJSON(`${LICENSE_CATEGORIES_LOCATION}/${category}.json`)
+async function fetchCategoryLicenses(categories) {
```
Comment (giannif) [inline scripts/license-checker.js:29]: @pshoniuk I'm a little lost, how is this script actually invoked?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -29,4 +29,4 @@ jobs:
               run: yarn run check-licenses
```
Comment (giannif) [inline .github/workflows/main.yml:29]: @pshoniuk if we're running this, then what does the licenses.yml do?

File: .editorconfig (hunk 1 above)
Comment (pshoniuk) [reply .editorconfig:12]: This configuration doesn't conflict with prettier rules. EditorConfig helps maintain common settings between different IDEs. For example, I use WebStorm, and my default setting defines two spaces for yaml files, but this project uses 4

File: scripts/license-checker.docker.sh (hunk 2 above)
Comment (pshoniuk) [reply scripts/license-checker.docker.sh:19]: Yes

**Review by giannif — APPROVED** [review]
Check those licenses! 
<div><br/>via <a href="https://giphy.com/disneychannelofficial/">Disney Channel</a> on <a href="https://giphy.com/gifs/disneychannelofficial-4xWGyVKoXqg2eVCiq9">GIPHY</a></div>

**Pull request merged**